packages/astro/src/core/sync/index.ts

```diff
@@ -251,7 +251,24 @@ async function syncContentCollections(
 				ssr: { external: [] },
 				logLevel: 'silent',
 			},
-			{ settings, logger, mode, command: 'build', fs, sync: true, routesList, manifest },
+			{
+				settings: {
```

Comment:
You might want to leave a longer comment that explains why we're doing this.

Also, we want to explain why some properties are overridden while some others aren't.

I believe it's very important

Reply:
```suggestion
				// Prevent mutation of specific properties by vite plugins during sync
				settings: {
```